apps/studio/src/features/editing-experience/hooks/useLinkEditorModal.tsx

```diff
@@ -0,0 +1,47 @@
+import type { Editor } from "@tiptap/react"
+import { useDisclosure } from "@chakra-ui/react"
+
+import { LinkEditorModal } from "~/components/PageEditor/LinkEditorModal"
+
+interface UseLinkEditorModalProps {
+  editor: Editor
+}
+
+export const useLinkEditorModal = ({ editor }: UseLinkEditorModalProps) => {
+  const {
+    isOpen: isLinkModalOpen,
+    onOpen: onLinkModalOpen,
+    onClose: onLinkModalClose,
+  } = useDisclosure()
+
+  return {
+    onLinkModalOpen,
+    LinkEditorModal: (
+      <LinkEditorModal
+        linkText={
+          editor.isActive("link")
+            ? editor.state.doc.nodeAt(
+                Math.max(1, editor.view.state.selection.from - 1),
+              )?.textContent
+            : ""
+        }
+        linkHref={
+          editor.isActive("link") ? editor.getAttributes("link").href : ""
+        }
+        onSave={(linkText, linkHref) => {
+          editor
+            .chain()
+            .focus()
+            .extendMarkRange("link")
+            .deleteSelection()
+            .insertContent(`<a href="${linkHref}">${linkText}</a>`)
+            .run()
```

Comment:
how does this work with text content with 2 marks? a valid use case would be, for example, to make a link bold!

i think this might delete all other marks in this case. maybe a more appropriate command might be `toggleNode`? (see [here](https://tiptap.dev/docs/editor/api/commands/nodes-and-marks/toggle-node))

Reply:
Hmm this one I'm not very sure. I went with this option because this seems to be the only way to change the text of the original link with the value of the new link text provided by the user.

Alternative is to build an extension that can handle this for us using ProseMirror's APIs directly, I can maybe follow this up on a separate PR?